src/index.ts

```diff
@@ -1,87 +1,80 @@
-    symbol?: string
-    url?: string
-}
-
-export interface SampleUsage {
-    account: API.v1.AccountObject
-    cpu: UInt128
-    net: UInt128
+interface TokenOptions {
+    apiClient: APIClient
+    systemTokenSymbol?: Asset.SymbolType
 }
 
-export const BNPrecision = new BN(1000 * 1000)
+export class Token {
+    readonly systemTokenContract: SystemTokenContract.Contract
+    readonly systemTokenSymbol: Asset.SymbolType
 
-export class Resources {
-    static __className = 'Resources'
+    constructor({apiClient, systemTokenSymbol}: TokenOptions) {
+        this.systemTokenContract = new SystemTokenContract.Contract({client: apiClient})
+        this.systemTokenSymbol = systemTokenSymbol || '4,EOS'
+    }
 
-    readonly api: APIClient
+    async transfer(
+        from: NameType,
+        to: NameType,
+        amount: number,
+        memo = '',
+        symbolType: Asset.SymbolType = this.systemTokenSymbol
+    ) {
+        const symbol = Asset.Symbol.from(symbolType)
+        const senderBalance = await this.balance(from, symbol.code)
 
-    // the account to use when sampling usage
-    sampleAccount = 'b1'
+        // check recipient balance to make sure that it exists
+        await this.balance(to, symbol.code)
 
-    // token precision/symbol
-    symbol = '4,EOS'
+        const quantity = Asset.from(amount, symbol)
 
-    constructor(options: ResourcesOptions) {
-        // Allow overriding of the sample account name
-        if (options.sampleAccount) {
-            this.sampleAccount = options.sampleAccount
-        }
-        // Allow overriding of the system token symbol
-        if (options.symbol) {
-            this.symbol = options.symbol
-        }
-        // Allow variations on how to specify the API configuration
-        if (options.api) {
-            this.api = options.api
-        } else if (options.url) {
-            this.api = new APIClient({provider: new FetchProvider(options.url, options)})
-        } else {
-            throw new Error('Missing url or api client')
+        if (senderBalance.value < amount) {
+            throw new Error(`Insufficient funds for ${from} to transfer ${quantity} to ${to}.`)
         }
-    }
 
-    v1 = {
-        powerup: new PowerUpAPI(this),
-        ram: new RAMAPI(this),
-        rex: new REXAPI(this),
+        return this.systemTokenContract.action('transfer', {
+            from: Name.from(from),
+            to: Name.from(to),
+            quantity,
+            memo,
+        })
     }
 
-    async getSampledUsage(): Promise<SampleUsage> {
-        const account = await this.api.v1.chain.get_account(this.sampleAccount)
-        const us = UInt128.from(account.cpu_limit.max.value.mul(BNPrecision))
-        const byte = UInt128.from(account.net_limit.max.value.mul(BNPrecision))
+    balance(accountName: NameType, symbolCode?: Asset.SymbolCodeType): Promise<Asset> {
+        return this.systemTokenContract
+            .table('accounts', accountName)
+            .all()
```

Comment:
I think you can specify an index here to get a specific token, instead of the logic you're using below to iterate and find it from all results. Might as well not ask for the data if we're not going to use it.

Reply:
Yeah makes sense! Will try it out!